### DISCUSSION TITLE
GS-T3E v1 - DISCONTINUED PRODUCT !!

### DISCUSSION THREAD

**Discussion by htvekov** [HASwitchPlate/openHASP#383] [Hardware]
# DISCONTINUED DEVICE - NEW T3E v2 SKU AVAILABLE
# CHECK [THIS](https://github.com/HASwitchPlate/openHASP/discussions/458) DISCUSSION THREAD

### This thread is made to gather all possible info on the GS-T3E device with the goal to get openHASP implemented.

![T3E_DEVICE](https://user-images.githubusercontent.com/45469813/190478532-135983f7-294f-4c61-981f-6ef2721ee6e2.jpg)

[Alibaba Link](https://www.alibaba.com/product-detail/2022-new-smart-home-tuya-switch_1600560503013.html)

Link to various [pictures](https://www.dropbox.com/sh/mtx2rxuit07tfe3/AACAtnrmkDsp_gQBC5Fn0N-wa?dl=0) of the device.

This device is in exterior completely identical to the WT86-32 device, as the outer casing and dimensions are 100% identical.
Device is marketed as an in wall multi-function scene panel and a 3(or 4) gang relay switch. The [T3E](https://www.dropbox.com/s/9li72bxs8s0zvjk/T6E%20VS%20T8E%20and%20T3E.pdf?dl=0) is one of three devices in a series. Both the [T6E & T8E](https://www.dropbox.com/s/9li72bxs8s0zvjk/T6E%20VS%20T8E%20and%20T3E.pdf?dl=0) devices have Tuya zigbee gateway capabilities, where the T3E can only act as a 'slave' device and smart scene panel display. These devices can only be used with the 'Smart Life' / 'Tuya Smart' apps. No local control - only cloud based control through these apps.

**EU backplate version:**
- 3 gang relay switch
  * An EU backplate 1 gang relay switch + 1 way light dimming version should also be available (0-10V Dimming Wiring Port)
- Max load: 200W pr. gang (600W in total EU version)
- Dimensions: 86x86x36mm
- Device pcb version: v1.23 (20220817)

**T3E compared to WT86-32:**

**PROS:**
- 4" Capacitive HD IPS LCD
  * Resolution: 480 x 480
  * Display driver: [ST7701S](https://ronboelectronics.en.made-in-china.com/product/kdumLsbPArWB/China-Smart-Home-86-Box-4-0-Inch-Resolution-480-X-480-IPS-Capacitive-Touch-Square-LCD-Display.html)
  * Touch controller: Goodix [GT911](https://www.distec.de/fileadmin/pdf/produkte/Touchcontroller/DDGroup/GT911_Datasheet.pdf)
- CPU: [ESP32-S3R8](https://www.espressif.com/sites/default/files/documentation/esp32-s3_datasheet_en.pdf) (Xtensa-LX7 dual core 240MHz)
  * 8 MB PSRAM - In package
  * GigaDevice 128mb flash (16 MB) - External flash
- EU backplate
- Micro USB power connector

**CONS:**
- Tuya device with 'standard' ZS3L Tuya chip on pcb
- ESP32 antenna IPEX connector **_not_** populated on pcb
- ESP32 flash header pins **_not_** populated on pcb  

**`U13` EU backplate connector (marked with `yellow` in picture below):**
Numbered from right (marked with a white dot) to left
1. GPIO46 = L1 relay
2. GPIO45 = L2 relay
3. GPIO44 = L3 relay and `RX`
4. GPIO43 = L4 relay and `TX`
   (**Not** connected in the EU 3 gang relay switch version. L4 & L3 are low voltage outputs in the dimmer version)
6. GND
7. GND
8. VCC (+5V)
9. VCC (+5V)

**`J1` ESP32 flash header (marked with `red` in picture below):**
'Standard' flash header available on the pcb for easy flashing. The actual header pins are **not** populated, so you have to solder these on yourself.
NB ! The white pcb cover plate actually has holes that match the header pins, but only [short profile female header pins](https://www.aliexpress.com/item/10000000838267.html?spm=a2g0o.productlist.main.23.50907bf2A6jEcF&algo_pvid=99e9740c-944e-415a-8edc-e0a91448567a&algo_exp_id=99e9740c-944e-415a-8edc-e0a91448567a-11&pdp_ext_f=%7B%22sku_id%22%3A%2220000000003548910%22%7D) will fit under the cover plate. Standard Dupont male pins won't fit into these female headers. But the commonly used breadboard pins do fit. If you solder in 'regular' male header pins, the backplate can't be fitted, unless you modify the pins in height.

I flashed my device using a USB serial adapter CH340G and flash download tool v3.9.3.
Note that RX & TX are NOT cross wired !
`GND` <--> `GND`
`5V` <--> `5V`
`RX` <--> `RX`
`TX` <--> `TX`
`IO0` pulled low (to GND) during power on to enter boot mode.

**`K1` button (marked with `orange` in picture below):**
Connected to `EN` on the ESP32-S3 and acts as a `RESET` button

**`ANT1` IPEX antenna connector (marked with `purple` in picture below):**
The ANT1 is unfortunately not populated from supplier. Without adding an external antenna, the wifi quality is not sufficient, unless device is within a few meters from the AP. It's difficult, but not impossible, to solder in an [smd ipex connector](https://www.aliexpress.com/item/10000000804196.html?spm=a2g0o.productlist.main.5.3f3e6c9asQBtF1&algo_pvid=2878b5e8-a0fa-4d96-87c3-436936e76836&algo_exp_id=2878b5e8-a0fa-4d96-87c3-436936e76836-2&pdp_ext_f=%7B%22sku_id%22%3A%2220000000003410816%22%7D) on the pcb pads. As an alternative, an antenna can just be soldered directly on the pcb instead. That is a much simpler solder task.

With the connector and a 'standard' AliExpress [5Dbi gain flexible antenna](https://www.aliexpress.com/item/1005004141773367.html?spm=a2g0o.productlist.main.3.3f3e6c9asQBtF1&algo_pvid=2878b5e8-a0fa-4d96-87c3-436936e76836&algo_exp_id=2878b5e8-a0fa-4d96-87c3-436936e76836-1&pdp_ext_f=%7B%22sku_id%22%3A%2212000028170780009%22%7D) you'll get decent signal strength, but still with room for approvement. Signal strength is adequate to keep device connected without dropout issues at the furthest distance in my house from the AP.   

Note: C16 & C17 decoupling capacitors are also not populated. This has also been noticed as well on other similar pcb designs where ESP antenna actually is part of the design. So perhaps not that important after all. I haven't experienced any wifi dropouts or quality issue with my external antenna, even though C16 & C17 are left unpopulated.

**`U2` connector/pads (marked with `indigo` in picture below):**
VCC: +3.3V
GND: GND
RXD: GPIO48
TXD: GPIO47
NRST: NC (no connection) 

**`P1` USB Micro-B connector (marked with `rosa` in picture below):**
1. VCC: +5V
2. D-: GPIO19
3. D+: GPIO20
4. ID: NC (no connection)
5. GND: GND

**Other useful pin connections identified:**
`GPIO38` = Display backlight (`LCD_BL`)

![190471634-4ae59542-c25a-4bef-8be6-04283ad887fc](https://user-images.githubusercontent.com/45469813/192022414-56239e70-be80-4415-a955-71d8123354b6.jpg)

**Comment by htvekov** [comment]
Current status: Tasmota 'standard' downloadable binaries running smoothly. No issues observed
Display status: Conducting tests with the Arduino_GFX library and examples: https://github.com/moononournation/Arduino_GFX/blob/master/examples/PDQgraphicstest/PDQgraphicstest.ino

**Comment by htvekov** [comment]
With (a lot of) help from 'puppet master' 😆 @fvanroie, I managed to get the pcb traces sorted correctly and the T3E device running the [Arduino GFX library]( https://github.com/moononournation/Arduino_GFX/blob/master/examples/PDQgraphicstest/PDQgraphicstest.ino) graphics test 🎉

![image](https://user-images.githubusercontent.com/45469813/190902281-5ce9118d-ab8f-45c4-83a1-c98178c03c4c.png)

Below the working pin configuration:

```
#elif defined(ESP32_S3_RGB)
#define TFT_BL 38
// Note. BO & B1 share ESP pin connections with SDA & SCK !!
Arduino_ESP32RGBPanel bus = new Arduino_ESP32RGBPanel(
    39 / CS /, 15 / SCK /, 14 / SDA /,
    18 / DE /, 17 / VSYNC /, 16 / HSYNC /, 21 / PCLK /,
    4 / R0 /, 3 / R1 /, 2 / R2 /, 1 / R3 /, 0 / R4 /,
    10 / G0 /, 9 / G1 /, 8 / G2 /, 7 / G3 /, 6 / G4 /, 5 / G5 /,
    15 / B0 /, 14 / B1 /, 13 / B2 /, 12 / B3 /, 11 / B4 /
);
Arduino_ST7701_RGBPanelgfx = new Arduino_ST7701_RGBPanel(
  bus, GFX_NOT_DEFINED /* RST /, 0 / rotation /,
  true / IPS /, 480 / width /, 480 / height /,
  st7701_type1_init_operations, sizeof(st7701_type1_init_operations),
  true / BGR */);

```

**Reply by fvanroie** [reply]
This is the ESP32-S3 pin usage: *(click to zoom in)*
![image](https://user-images.githubusercontent.com/15969459/190923192-18d02f65-817b-42bf-9bc0-d24aee3b8d3b.png)

**Reply by fvanroie** [reply]
Some updates:
- I contacted the seller about a WiFi version of this device, and they expect to have one by the end of the year. 🤞 
- The ST7701S display chip is only supported in the Arduino_GFX library, or a specific driver needs to be implemented for openHASP

**Reply by dmsherazi** [reply]
@fvanroie would it be possible to share the contact details of the manufacturer of this device?

**Reply by htvekov** [reply]
> @fvanroie would it be possible to share the contact details of the manufacturer of this device?

The initial vendor we had contact with is Global Security. Now there are numerous T3E vendors out there.
Here's the Alibaba link to the T3E device. Just message Maggie Tang from that page.
I expect the WiFi version of the T3E will be available shortly. No specs yet though from GS.
Hopefully they haven't completely revised the pcb layout or even worse changed mcu from ESP32-S3 to something odd...

https://www.alibaba.com/product-detail/2022-new-smart-home-220V-wall_1600560432462.html?spm=a2700.galleryofferlist.normal_offer.d_image.74b562a7xzasIQ

**Comment by nagyrobi** [comment]
> * ESP32 header pins **_not_** populated on pcb

Does this mean there's an empty location on the board to solder on an ESP32 module perhaps?

**Reply by htvekov** [reply]
Nope. The flash header **is** available on the PCB, but not populated with pins from supplier.

![image](https://user-images.githubusercontent.com/45469813/190926369-e77af41c-9489-4dba-8b5b-04497659a05d.png)

The pcb already has an ESP32-S3 onboard, so I don't really understand your question ? 🙂

**Reply by nagyrobi** [reply]
_ESP32 header_ vs _flash header_ I guess that wss confusing for me...

**Comment by htvekov** [comment]
T3E fitted with low profile female flash pin headers, Ipex connector and a flexible antenna.

![IMG20221030205647](https://user-images.githubusercontent.com/45469813/199128417-3edcc0df-d7d3-4968-87c0-1a529d599600.jpg)

**Comment by fvanroie** [comment]
Making progress with this device: I've got an openHASP display driver prototype working for this board.
The Touch driver still needs some work though...

**Reply by fvanroie** [reply]
![image](https://user-images.githubusercontent.com/15969459/201236339-4ada069b-4c47-4a6c-823b-3b6497e56aad.png)

**Reply by htvekov** [reply]
Looks great ! 🙂Really looking forward for the touch driver part 😉

**Comment by nagyrobi** [comment]
How does the ESP32-S3 in this device compare to good old ESP32? Does it have enough PSRAM? Will we be able to display complex pages, with animated objects like we can on Lanbon L8 for example? Or will TTF fonts work?

**Reply by htvekov** [reply]
> How does the ESP32-S3 in this device compare to good old ESP32? Does it have enough PSRAM? Will we be able to display complex pages, with animated objects like we can on Lanbon L8 for example? Or will TTF fonts work?

Hard to tell, as the current display driver is still a prototype and the touch hasn't been implemented yet. Page drawing atm. is quite slow with current driver config. Looking at other devices, I don't believe it's the actual ESP type that makes the big difference. More likely the display type used in conjunction with the driver. I guess that the octal or quad hw config most likely also plays a role.

Looking at the WT32-sc01 devices, there a HUGE difference in overall percention of speed between the 'old' ESP32 Wrover B version compared to the new plus device equipped with ESP32-S3. I guess most likely due to the lightning fast display update on the new plus device, rather than the actual ESP core speed. @fvanroie can most likely elaborate on this issue 🙂

PSRam is good at 8MB and external flash size is 16MB. So plenty of space...

**Reply by htvekov** [reply]
> > How does the ESP32-S3 in this device compare to good old ESP32? Does it have enough PSRAM? Will we be able to display complex pages, with animated objects like we can on Lanbon L8 for example? Or will TTF fonts work?
> 
> Hard to tell, as the current display driver is still a prototype and the touch hasn't been implemented yet. Page drawing atm. is quite slow with current driver config. 

Perhaps, display driver actually performs quite well. Noticed that page updating through the web client was extremely slow compared to commands

**Reply by fvanroie** [reply]
Please be aware that ESP-S3, ST7001S and Arduino_GFX support are all quite new and still need tweaking and proper testing.

> Noticed that page updating through the web client was extremely slow compared to commands

Don't use the Screenshot page or performance will be severely impacted. The full bitmap gets transferred over the network and the WiFi signal of the GS-T3E isn't very good. You can also switch pages from the File Editor which should be fast.

> Looking at the WT32-sc01 devices, there a HUGE difference in overall percention of speed between the 'old' ESP32 Wrover B version compared to the new plus device equipped with ESP32-S3. I guess most likely due to the lightning fast display update on the new plus device, rather than the actual ESP core speed.

A large part can be explained by the 8-bit parallel display on the WT32-SC01 Plus, which makes a big difference over SPI... The ESP32-S3 will *still* help since it *is* faster due to flash and PSram upgrades. There is also more free Sram available, because the S3 has more functions in ROM and needs less *hot patches* in system memory.

> How does the ESP32-S3 in this device compare to good old ESP32?

I would say the apparent performance is comparable or better, which is very good considering the display is pushing 3x the amount of pixels. The GS-T3E is using a 16-bit parallel bus to drive the display, where the regular ESP32 uses SPI or *in rare cases* 8-bit.

> Does it have enough PSRAM?

This GS-T3E has 8 MiB of **usable** PSram, which is double the usable size in the ESP32. Also the PSram is twice as fast on this S3 (octal vs quad speed).

> Will we be able to display complex pages, with animated objects like we can on Lanbon L8 for example?

I don't see a reason why it shouldn't be able to handle the same workload, maybe even a little more. We can load the pages.jsonl on the GS-T3E and see how it performs...

> Or will TTF fonts work?

Yes, the example is using TTF fonts for the icons and clock.

> Really looking forward for the touch driver part.

Touch is working but with some hacks in the driver which need to be cleaned up.

**Reply by nagyrobi** [reply]
Thank you for the details. Do you think this will be an officially supported device?

**Reply by htvekov** [reply]
> > Really looking forward for the touch driver part.
> 
> Touch is working but with some hacks in the driver which need to be cleaned up.

Best sentence I've read so far today! 💪😎

**Reply by htvekov** [reply]
> I would say the apparent performance is comparable or better, which is very good considering the display is pushing 3x the amount of pixels. The GS-T3E is using a 16-bit parallel bus to drive the display, where the regular ESP32 uses SPI or _in rare cases_ 8-bit.

I guess the T3E will be faster than the future WT86-32-SZ01 SKU (if ever released) as the specs for that device is a 8-bit MCU (8080). So that device will most likely struggle quite a bit moving same amount of data to the screen (also 480x480 res.) than the T3E ?

**Reply by htvekov** [reply]
> Thank you for the details. Do you think this will be an officially supported device?

Hopefully, yes. @fvanroie probably has the very first and only device here in western Europe equipped with an ipex connector, external antenna, low profile header pins and a matching 'quick flash jig' 😁

But I guess that future 'official' device support will be closely related to the device wifi capabilities IRL -> production environment.

**Reply by fvanroie** [reply]
Thanks for that! 😺   The missing ipex connector and sub-optimal WiFi signal are some of the drawbacks to make it into an "officially supported device". I guess its use will be limited to the hardware hackers and soldering wizards like @htvekov .

The reason for this effort is because there *should* be some comparable WiFi devices in the near future that will make for much better candidates... But we know that Chinese seller will say about anything, so untill those devices actually materialize this is the only hardware we have atm.

There seems to be an influx of new ESP32-S3 display boards on the horizon with larger display sizes, so it's better to prepare in advance...

**Comment by nagyrobi** [comment]
Don't see on Aliexpress... Where advised to buy?

**Reply by htvekov** [reply]
> Don't see on Aliexpress... Where advised to buy?

All device specs and Alibaba link in first tread entry. Not available on AliExpress - yet.
Be aware that ipex connector and flash download header pins are not soldered in. There are link examples to those as well.
Also note that even with external antenna fitted via ipex connector, wifi strength is only acceptable and not as good as expected.
No real 'in production' tests has yet been done, as @fvanroie is still tweaking device config - especially the touch part

A 'proper' wifi version of the device should be available within this year, but nothing yet confirmed from supplier.

**Reply by fvanroie** [reply]
There is [this link](https://www.alibaba.com/product-detail/2022-new-smart-home-tuya-switch_1600560503013.html) on Alibaba, and some [other](https://www.alibaba.com/product-detail/T3E-2022-new-Wall-Switch-smart_1600604850500.html) [vendors](https://www.alibaba.com/product-detail/2022-new-design-tuya-smart-home_1600540410320.html) are also selling the T3E under their own brand, so it is likely a rebranded OEM device.

I've asked about a WiFi version and the first seller (Shenzhen Golden Security Technology) replied that it is in the works... Let's hope this is actually true. The more people ask about WiFi versions instead of the Tuya $#[£ then they might see the need for such a device for the European market.

**Comment by htvekov** [comment]
First 'useful' plate to utilize the T3E's 4" 480x480 res. almost ready designed. Took like forever...
Solar energy plate only missing some connecting lines, and logic that changes the spinners speed and arc angle according to values and the mix of energy consumed from grid and solar panels.

Tweaked HA's Energy component design to fit the LVGL possibilities in openHASP.

![image](https://user-images.githubusercontent.com/45469813/203407590-56b885cf-1f36-4d27-b3cb-058159011d06.png)

**Comment by htvekov** [comment]
Sonos media plate, utilizing the large 480x480 res. display, also mostly redesigned and done.
Progress bar moved in between title/artist objects and all objects lying on top of the image with varying opacity

![image](https://user-images.githubusercontent.com/45469813/205514702-180bb8a4-02c5-47c9-b216-8efdb21da39d.png)

**Comment by dmsherazi** [comment]
HI All , First of all I would thank you all for all these efforts. 
Secondly I have this board and woulld like to work on it, I tried to get some code examples  from a similar board. https://github.com/W00ng/ESP32-S3-RGB-Panel and tried to flash an example on it, It did got flash but there is no display on the screen. May be the board has diffrent pin configs or I am doing something wrong . Can you please share your bin file for the T3E so I can test?

**Reply by fvanroie** [reply]
You can download the artifact from the latest workflow [Actions](https://github.com/HASwitchPlate/openHASP/actions) or flash it directly from the [development web tools](https://nightly.openhasp.com).

**Reply by htvekov** [reply]
> HI All , First of all I would thank you all for all these efforts. Secondly I have this board and woulld like to work on it, I tried to get some code examples from a similar board. https://github.com/W00ng/ESP32-S3-RGB-Panel and tried to flash an example on it, It did got flash but there is no display on the screen. May be the board has diffrent pin configs or I am doing something wrong . Can you please share your bin file for the T3E so I can test?

Well, @fvanroie has made a tailor fit, downloadable binary from the Action flows.
Artifact name is globalsecurity (first vendorname for that device) and not T3E if you were looking for that.

If that binary doesn't work for you, then please check the pcb version on your device and compare to the pcb version found in pictures in the start of this thread (Device pcb version: v1.23 (20220817))

**Reply by dmsherazi** [reply]
Thanks for the help and it did work. Even the . https://github.com/W00ng/ESP32-S3-RGB-Panel  examples also now are working .. Thanks to the details and information provided in this discussion.

I have a question : Is this a 32bit display , or a 16bit ?

**Reply by htvekov** [reply]
> Thanks for the help and it did work. Even the . https://github.com/W00ng/ESP32-S3-RGB-Panel examples also now are working .. Thanks to the details and information provided in this discussion.
> 
> I have a question : Is this a 32bit display , or a 16bit ?

No worries 😎
The T3E device has a 16bit display (rgb565)

**Comment by htvekov** [comment]
All needed files for my openHASP multiple plate - Sonos group media player config have now been revised and uploaded [here](https://github.com/htvekov/openHASP-Sonos-media-player)

Documentation still needs to be revised somewhat from my old config setup.
For now only prerequisities has been revised.
Note: When doing entity_id search and replace in config files, search for t3e_02 instead.

I'll get a decent documentation done ASAP 😅

![image](https://user-images.githubusercontent.com/45469813/213568033-0b6860d6-e289-4365-95d2-ae06a5c696bd.png)

**Reply by htvekov** [reply]
Documentation complete now 🙂

**Comment by shavermcspud** [comment]
I have a version 2.3 board, flash has worked fine, configured the WiFi but have no display, I can access the web gui just fine. 
Any thoughts

**Reply by htvekov** [reply]
> I have a version 2.3 board, flash has worked fine, configured the WiFi but have no display, I can access the web gui just fine. Any thoughts

Hard to tell, as display pin connections could have been altered or display driver revised since v1.23
Could you upload a close-up high-res picture of the pcb ?

**Reply by dmsherazi** [reply]
Can you share the images of the PCB?

**Comment by shavermcspud** [comment]
To add, it was purchased from golden security and is listed as a GS-T3E


![PCB v2 3](https://user-images.githubusercontent.com/96439393/223344751-a35effc2-c090-4016-8223-f00a9cea145b.jpg)

**Reply by fvanroie** [reply]
This board has an ESP32-S3 as module, which is significantly different from the previous PCB version.
It has the potential for WiFi communication too, so that is hopeful.
We need a pinmapping of the traces like [my post above](https://github.com/HASwitchPlate/openHASP/discussions/383#discussioncomment-3675684) in order to compare/create the LCD pins.

Where did you buy this device with PCB 2.3 ?

**Reply by shavermcspud** [reply]
Hi, I got it from Alibaba, golden security, I was under the impression speaking with the seller it was the version listed above and not 2.3 revision, unfortunately I have bought quite a few of them and its a little above my level of knowledge to map things out.

**Comment by htvekov** [comment]
@shavermcspud & @fvanroie

Well, this is crazy as both me and fvanroie has been banging this vendor (Global Security) about the T3E Wifi version availability and now it seems to have been available for months as the pcb clearly **_has_** been redesigned 🙄 I really don't care whether or not the pcb is still populated with a Tuya chip or not.

But new pinmapping apparently is needed if display is not working with the new pcb v2.3
Please check the 40pin display FFC cable and check if it's still same part no. marked with `BH040I-01` and populated with a Goodix GT911 touch  chip in the corner.

![image](https://user-images.githubusercontent.com/45469813/223491019-4f25093a-8433-401b-ae42-60178e2dca81.png)

**Reply by dmsherazi** [reply]
It seems to me that Global security might not be the manufacturer.  @shavermcspud  can you let us know from where you got it?

**Reply by htvekov** [reply]
> It seems to me that Global security might not be the manufacturer. @shavermcspud can you let us know from where you got it?

Sorry my bad. I _meant_ to write 'Golden Security' not 'Global Security'
@shavermcspud already stated that is was bought from GS (text above the image)

**Comment by shavermcspud** [comment]
Can confirm the touch is populated with a Goodix GT911 and the ribbon cable shows BH040I-01, see photos
Purchased from Alibaba Global Security three weeks ago, unit flashed just fine with your T3E bin file, WiFi works unit is programed and seen in HA as a plate.
![plate](https://user-images.githubusercontent.com/96439393/223501718-fb3723a0-e0d5-49c8-8127-c5e870613d31.jpg)
![ribbon](https://user-images.githubusercontent.com/96439393/223501723-a5aa67f5-80e3-4d56-96c5-fe5bfac81bf5.jpg)
![GT911](https://user-images.githubusercontent.com/96439393/223501733-a9a7113c-e254-4ad9-ac88-36ca2c282d0e.jpg)

**Comment by htvekov** [comment]
Well, seems like the display/touch part is unrevised 😅😅
Is backlight working correctly ?
You can also check if touch is working. Just touching the display should change mode from idle to off on the info web interface page.

Seems that you'll need to check the traces to the display as @fvanroie wrote earlier. Start with all other traces than the actual color bits (DB0 to DB17). If backlight and touch is working these pins below are the most interesting ones

![image](https://user-images.githubusercontent.com/45469813/223513077-23646ba0-4e21-4770-aace-65d8513bf32b.png)

**Reply by shavermcspud** [reply]
That's the issues, Backlight isn't on, touch changed from idle off to idle short so I don't think that's working either.

**Reply by htvekov** [reply]
Well, as far as I can tell from your picture backlight pin (GPIO38) is connected to HS (connector pin 15)
And if touch is not working either, they might have swapped multiple pins around on this new pcb layout.
There no easy fix to this than tracing each and every pin connection (visually or using a multimeter) and compare to @fvanroie pin trace image above.

I've only managed to find a pin layout for the N8R8 (Hard to tell the type on your picture. Should be an MON16R8)
But pins should be identical AFAIK

![ESP32S3_MON8R8](https://user-images.githubusercontent.com/45469813/223521828-c2a5748d-7650-41ef-9c63-a8158a9a2e52.png)

**Reply by shavermcspud** [reply]
I think touch is working ok, if i press the display, it does change fro idle short/long to off so that seems to be good. Found the pinout and its exactly as yours above. 
![wroom](https://user-images.githubusercontent.com/96439393/223525400-1f3634db-828b-47e8-9a79-f6e8922f3b31.jpg)

**Reply by htvekov** [reply]
> I think touch is working ok, if i press the display, it does change fro idle short/long to off so that seems to be good. Found the pinout and its exactly as yours above.

Well, you need to trace/measure the important GPIO's then for driving the display.
Forget the top seven pins as they control the touch part. And don't worry about the RGB bits for now. Focus on the first 15 pins.
Traces are not visible from the top of the pcb. So I would measure with a multimeter 🙂

**Reply by shavermcspud** [reply]
I think ill just pop them in the bin, I have zero chance of being able to do that, I can barley see my keyboard :-) let alone run my meter across those pins on the flex connector.

**Reply by htvekov** [reply]
> I think ill just pop them in the bin, I have zero chance of being able to do that, I can barley see my keyboard :-) let alone run my meter across those pins on the flex connector.

Well, hang on to those devices for a bit, as I'm about to reorder some T3E devices. Was actually just waiting for GS to confirm the WiFi version was available before ordering. Clearly it's already there, despite the populated Tuya chip that doesn't really matter.
I'll sort the pin config with @fvanroie when I get my devices delivered (hopefully)
Did you order with EU- or Chinese sized end plate ?

**Reply by htvekov** [reply]
You could identify the backlight pin by simple trial and error iterating through different BL pins in the web interface.
Yould could end up crashing the plate though, so you'll need to reflash via serial.

**Reply by fvanroie** [reply]
Most pins are very straightforward to map because the trace goes straight into the ESP.
There's a good chance it is GPIO42 but needs to be verified.

The pins that lead to a VIA are unknown at this time. Maybe it is possible to get a high res images of the backside of the PCB too?

**Reply by shavermcspud** [reply]
> You could identify the backlight pin by simple trial and error iterating through different BL pins in the web interface. Yould could end up crashing the plate though, so you'll need to reflash via serial.

I did try that, wasn't sure if I needed to restart the plate after every pin change but i did that anyway, set pin 1 restart and so on, it crashed when set from 22 to 27 and I had to re flash it several times

GPIO42 is indeed the backlight, that is now on and working. had to go through every one.

**Reply by shavermcspud** [reply]
![image](https://user-images.githubusercontent.com/96439393/223556273-2bf94e2d-c642-4bc2-9b15-63219b20451d.png)

Here so far with probing, is it normal to have that many not connected?

**Reply by htvekov** [reply]
Yep. Pin 6,7 & 8 are NC 👍
I believe you should/could stop now, as I'm fairly sure @fvanroie has sorted 98% of this already 🙂
Perhaps a pin or two is missing still ?

**Reply by fvanroie** [reply]
Only 3 pins left, but I can kinda guess:

| # | Function | GPIO
-- | :--: | --:
40 | TP_GND | GND
39 | TP_VDD | VDD
38 | TP_RST | NC
37 | TP_SCL | 6 ?
36 | TP_SDA | 7 ?
35 | TP_INT | NC
34 | GND | GND
33 | DB17 | 0
32 | DB16 | 16
31 | DB15 | 17
30 | DB14 | 18
29 | DB13 | 8
28 | DB12 | GND
27 | DB11 | 3
26 | DB10 | 9
25 | DB9 | 10
24 | DB8 | 11
23 | DB7 | 12
22 | DB6 | 13
21 | DB5 | 14
20 | DB4 | 21
19 | DB3 | 47
18 | DB2 | 48
17 | DB1 | 15 ?
16 | DB0 | GND
15 | HS | 38
14 | VS | 39
13 | DE | 40
12 | DCLK | 41
11 | CS | 4
10 | SCK | 15
9 | SDA | 48
8 | NC | NC
7 | NC | NC
6 | RESET | NC
5 | VSS | VSS
4 | GND | GND
3 | LED_K2 | GND
2 | LED_K1 | GND
1 | LED_A | 42
USB | DP+ | 19
USB | DP- | 20

**Reply by htvekov** [reply]
Touch was working, so the TP pins are correct I guess ?

**Reply by fvanroie** [reply]
I will be very surprised if touch is indeed working. The old I2C pins where `40` and `41` which are now used for the display.
None of the missing pins are critical to get the display working. We can double-check the rest later.

**Reply by shavermcspud** [reply]
The only reason i say the touch might be working, is from the post above, the information display shows touch long, ive left the plate on all night, I just touched the display and its gone to idle off

**Reply by shavermcspud** [reply]
> > I think ill just pop them in the bin, I have zero chance of being able to do that, I can barley see my keyboard :-) let alone run my meter across those pins on the flex connector.
> 
> Well, hang on to those devices for a bit, as I'm about to reorder some T3E devices. Was actually just waiting for GS to confirm the WiFi version was available before ordering. Clearly it's already there, despite the populated Tuya chip that doesn't really matter. I'll sort the pin config with @fvanroie when I get my devices delivered (hopefully) Did you order with EU- or Chinese sized end plate ?

I have a spare unit with a broken screen, happy to post over to you to mess with, as the the display is the same.

**Reply by htvekov** [reply]
> The only reason i say the touch might be working, is from the post above, the information display shows touch long, ive left the plate on all night, I just touched the display and its gone to idle off

Hmm... Somewhat surpricing (unexplainable) that touch should be working at all, as @fvanroie wrote that TP_SDA & TP_SCL both have new i2c pin assignments 🤔

**Reply by shavermcspud** [reply]
Its only a guess, as the info screen shows somthing, but this could be random

**Reply by htvekov** [reply]
> I have a spare unit with a broken screen, happy to post over to you to mess with, as the the display is the same.
 
 Where are you located ? I live in Denmark, so perhaps the shipping cost will be horrible and not worth it ?
 My new T3E order hasn't shipped yet, so I won't recieve anything directly from GS before early April.



> Its only a guess, as the info screen shows somthing, but this could be random

Well, if idle changes in web interface upon touch and otherwise stays stable, _something_ is working regarding touch  even if it shouldn't

**Reply by shavermcspud** [reply]
I am in the UK, don't think postage of a single board would be too much.=, is there a way to private message your address to me?

**Reply by cook1emr** [reply]
> I am in the UK, don't think postage of a single board would be too much.=, is there a way to private message your address to me?

How well do they fit a UK back box?

**Reply by htvekov** [reply]
> I am in the UK, don't think postage of a single board would be too much.=, is there a way to private message your address to me?

Just pinged you via Discord

**Reply by shavermcspud** [reply]
> > I am in the UK, don't think postage of a single board would be too much.=, is there a way to private message your address to me?
> 
> How well do they fit a UK back box?

Its quite tight with the four gang switch on the back but does fit.

**Reply by shavermcspud** [reply]
@htvekov I have responded on Discord.  Dumb question maybe from uneducated me, whats the process to change the current firmware io pin assignment for the new pinout ?

**Reply by fvanroie** [reply]
Are you familiar with Arduino? The first step is to compile a default sketch to get the display working with Arduino_GFX.
https://github.com/HASwitchPlate/openHASP/discussions/383#discussioncomment-3674155

**Reply by htvekov** [reply]
> Are you familiar with Arduino? The first step is to compile a default sketch to get the display working with Arduino_GFX. [#383 (comment)](https://github.com/HASwitchPlate/openHASP/discussions/383#discussioncomment-3674155)

Yep, I agree. Easiest way to check pin config with device. Also quite easy to revise pin config and recompile quickly

**Reply by shavermcspud** [reply]
Unfortunately not, its a little beyond my skill level. The last interaction I had with Arduino was for my 3D printer to re-flash the filament cartridge to accept 3rd party filament. But that was a pre made sketch from memory a long time ago.  For the $30 these units cost me, they seemed like a great addition to finish off my Home assistant setup, but not  sure the vendor was 100% honest with their sales pitch as these are clearly different units to the ones I had wanted.

**Reply by htvekov** [reply]
I can mock-up that pin config in the sketch for you to compile. Just have to locate sketch again. Haven't used it since last fall 😆
A few traces are still unknown. Could you provide a high-res picture of the pcb backside ?
@fvanroie has traced all connections that is visible from your picture, but a few would be pure guessing without that flipside image

**Reply by htvekov** [reply]
As i can't upload INO file types here, I've instead added a link to the test Arduino GFX sketch for the T3E v2.3 pcb [here](https://www.dropbox.com/s/mdqddxkvwcrdxcd/T3E_test_sketch_pcb_v2.3.ino?dl=0)
I've configured according to @fvanroie trace config further up in this thread.
Two color bit have unknown traces, so they might be wrong. But sketch should run anyway and display fire up, unless I've made errors in the entries 😉

Give it a spin and let us know the result.

The pin configuration for the ESP32RGBPanel is from line 88 to line 97 in the INO file.

**Reply by shavermcspud** [reply]
Bottom PCB image 
![image](https://user-images.githubusercontent.com/96439393/224551966-a9da2907-88f4-40eb-b5ef-56b20c9011ed.png)

**Comment by htvekov** [comment]
Noticed that your WiFi signal strength was quite terrible at -82dBm ?
Is that value comparable with other esp devices you have close by ?
Original T3E device had WiFi issues as no antenna was on the pcb and no IPEX connector populated.
But this is now a ESP32 module with built-in antenna, so I would expect 'normal' signal strength with this

**Reply by shavermcspud** [reply]
My WiFi point is quite far away unfortunately

**Reply by shavermcspud** [reply]
Just to add, I have re-enabled a new access point in my computer room about 20 meters away and now the WiFi is good to the plate 


Wifi
--
BSSID | B8:EC:A3:DF:C3:55
SSID | The Schads
Signal Strength | -44dBm (Excellent)
IP Address | 10.0.1.19
Gateway | 10.0.1.254
DNS Server | 1.1.1.1
MAC Address | F4:12:FA:E7:1D:B4
